File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int ans = 0;
        map<char,int> freq;
        for(int i = 0;i < s.size();i++){
            int start = i;
            while(start != s.size()){
                if(freq[s[start]] != 0)break;
                freq[s[start]]++;
                ans = max(ans,start - i + 1);
                start++;
            }
            freq.clear();
        }
        return ans;
        
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) == 0: # If input string is empty return 0
            return 0

        maxlen = -1  # Stores the maximum length of substring. Initialised with -1
        set_a = set()
        l = 0

        for r in range(len(s)):
            if s[r] in set_a:
                while l < r and s[r] in set_a:
                    set_a.remove(s[l])
                    l += 1
            set_a.add(s[r])
            maxlen = max(maxlen,r-l + 1)
            
        return maxlen
        